### ISSUE TITLE
Crash after upload completes

### ISSUE THREAD

**Issue by sebaFlame** [airdcpp-web/airdcpp-webclient#446]
### Current UI and client versions:

```
Application version
AirDC++w 2.11.4 x86_64
Web UI version
2.11.5
Web UI build date
March 11, 2023 5:53 PM
API version
1
API feature level
7
```

### Operating system:

```
# cat /etc/os-release
NAME=Gentoo
ID=gentoo
PRETTY_NAME="Gentoo Linux"
ANSI_COLOR="1;32"
HOME_URL="https://www.gentoo.org/"
SUPPORT_URL="https://www.gentoo.org/support/"
BUG_REPORT_URL="https://bugs.gentoo.org/"
VERSION_ID="2.13"
# uname -a
Linux jupiter 6.1.19-gentoo #1 SMP PREEMPT_DYNAMIC Fri Mar 17 19:01:05 CET 2023 x86_64 Quad-Core AMD Opteron(tm) Processor 2373 EE AuthenticAMD GNU/Linux
```

### Steps to reproduce the issue:

I had someone download a file from me and it crashed after the upload completed. The issue started happening after I added about 1.5TB of files.

### Screenshots

[backtrace.log](https://github.com/airdcpp-web/airdcpp-webclient/files/11200804/backtrace.log)

**Comment by maksis** [comment]
Are running a debug build? Looks like it hit a debug assertion.

**Comment by sebaFlame** [comment]
Yes, I rebuilt with symbols and the debug flag when I first encountered the issue and tried to reproduce with a proper backtrace.

**Comment by maksis** [comment]
I'm afraid that those are different crashes. Can you get a proper output if you build it without the debug flag? There are a few bad debug assertions that will cause issues.

**Comment by sebaFlame** [comment]
I rebuilt without the debug flag and let someone download. The issue did not occur. I'm now waiting on the actual issue to occur.

**Comment by sebaFlame** [comment]
I've let it run with gdb attached for about 2 days now and no crash... I did optimize the hash db after it first crashed, might this have solved it?

**Comment by maksis** [comment]
> I've let it run with gdb attached for about 2 days now and no crash... I did optimize the hash db after it first crashed, might this have solved it?

I guess that's possible. It could also be a really rare crash that may not even be related to the upload. Maybe it's better if you create a new issue if it crashes again?
